sdk/core/src/services/WalletStorageService.ts

```diff
@@ -371,15 +365,137 @@ export default class WalletStorageService {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+    const privateKeyBuffer = this._keysService.getOwnPrivateKey()
+
+    const hashedId = await this._platformEncryptionTools.computePersonalHash(privateKeyBuffer, credentialId)
+
+    let encryptedCredential: string
+    try {
+      const { body, status } = await this._vaultApiService.getCredential(token, storageRegion, hashedId)
+      if (status >= 400) {
+        throw new SdkErrorFromCode('COR-23', { credentialId })
+      }
+
+      encryptedCredential = body.payload
+    } catch (error) {
+      // should be deleted during migration to affinidi-vault Phase #2
+      const bloomToken = await this.authorizeVcBloomVault()
+      const credentialIndexToDelete = await this._findCredentialIndexById(credentialId)
+      const { body, status } = await this._bloomVaultApiService.getCredentials({
+        accessToken: bloomToken,
+        start: credentialIndexToDelete,
+        end: credentialIndexToDelete,
+        storageRegion,
       })
+      if (status >= 400 && body?.length > 0) {
+        throw error
+      }
+
+      encryptedCredential = body[0].cyphertext
     }
 
-    const credentialIndexToDelete = credentialIndex ?? (await this.findCredentialIndexById(id))
-    return this.deleteCredentialByIndex(credentialIndexToDelete)
+    const privateKey = this._keysService.getOwnPrivateKey()
+    const credential = await this._platformEncryptionTools.decryptByPrivateKey(privateKey, encryptedCredential)
+    return credential
+  }
+
+  public async deleteCredentialById(audienceDid: string, credentialId: string, storageRegion?: string) {
+    const token = await this._authorizeVcVault(audienceDid)
+    storageRegion = storageRegion || this._storageRegion
+
+    try {
+      const { status } = await this._vaultApiService.deleteCredential(token, storageRegion, credentialId)
+      if (status >= 400) {
+        throw new SdkErrorFromCode('COR-26', { credentialId })
+      }
+    } catch (error) {
+      // should be deleted during migration to affinidi-vault Phase #2
+      const bloomToken = await this.authorizeVcBloomVault()
+      const credentialIndexToDelete = await this._findCredentialIndexById(credentialId)
+      const { status } = await this._bloomVaultApiService.deleteCredentials({
+        accessToken: bloomToken,
+        storageRegion,
+        start: credentialIndexToDelete,
+        end: credentialIndexToDelete,
+      })
+
+      if (status >= 400) {
+        throw error
+      }
+    }
+  }
+
+  public async deleteAllCredentials(storageRegion?: string): Promise<void> {
+    const accessToken = await this.authorizeVcBloomVault()
+    storageRegion = storageRegion || this._storageRegion
+
+    try {
+      await this._bloomVaultApiService.deleteCredentials({ accessToken, storageRegion, start: 0, end: 99 })
+    } catch (error) {
+      throw new SdkErrorFromCode('COR-0', {}, error)
+    }
```

Comment:
Are we going to support `deleteAllCredentials` for affinidi-vault? If that is the case I would prefer to have batch delete on vault's side instead of having individual delete request for each credential here.

Reply:
There won't be batch delete on vault side, because it is impractical or impossible to do in a single transaction, especially for multi-region storage. Relevant discussion: https://lemmatree.slack.com/archives/C025J7PEG80/p1623679947068300?thread_ts=1623673672.040700&cid=C025J7PEG80

Handling of errors related to the incomplete delete should be done on a client side.
(And also turns out there was a bug in original `WalletStorageService` which remained in the updated version; it only deletes first 100 credentials from bloom vault.)

Reply:
okay, I see problem with batch delete described in that thread and here is method for delete all credentials, which means we don't want to pass list of items to delete, but want delete them all (in all regions), isn't it? We can remove this logic at all if we don't need it or rename it appropriately like `deleteNCredentials` or so?

Reply:
@DenisPopov15 @Yengas do you have some thought here?